Expanded module card UI

huzaifa1712 opened the pull request.

shenyih0ng commented:
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/47914880/196452038-6b34eb63-ceee-4f69-af80-1ee8d1a99493.png">

@huzaifa1712 I updated the `ModuleCard` UI by shifting all UI layout logic to `.fxml`, which means we no longer need to have `ListView` in `ModuleCard`

huzaifa1712 commented:
> > @huzaifa1712 I updated the `ModuleCard` UI by shifting all UI layout logic to `.fxml`, which means we no longer need to have `ListView` in `ModuleCard`
> 
> * Also, I fixed a bug regarding `Pill` where previously font sizes was not correctly updated.
> * Some UI changes, mostly for font sizes and also the color for "My Module" title

Thanks! Was having issues getting the text to wrap programatically. Didn't realise putting it in FXML would fix it.

shenyih0ng commented on src/main/java/nus/climods/model/module/Module.java:
```diff
@@ -71,6 +94,39 @@ public List<Integer> getSemesters() {
+        try {
+            if (detailedModule.isEmpty()) {
+                org.openapitools.client.model.Module module = modulesApi.acadYearModulesModuleCodeJsonGet(currentAcademicYear, this.getCode());
+                detailedModule = Optional.of(module);
```
I dont think we need a `Optional` wrapper since cmiiw the api throws a `404` when a module code is not found which our `try-catch` handles. We can probably simplify the logic without the use of `Optional`
